### PR TITLE
COST-2009 - Upgrade operator-sdk to v1.10.0

### PR DESCRIPTION
**Pull request by chambridge**
Made all migration updates documented here: https://sdk.operatorframework.io/docs/upgrading-sdk-version/
from v0.19.2 to v1.10.0.

Did not perform optional upgrade of kubebuilder mentioned here in v1.3.0 update: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.3.0/#upgrade-your-go-project-from-gokubebuilderiov2-to-gokubebuilderiov3

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -47,7 +47,7 @@ spec:
         volumeMounts:
         - mountPath: /tmp/koku-metrics-operator-reports
           name: koku-metrics-operator-reports
-      serviceAccountName: koku-metrics-manager-role
+      serviceAccountName: controller-manager
```
Comment (maskarb) [inline config/manager/manager.yaml:50]: Question: Why make this change?

I _think_ initially, these `serviceAccountNames` were all `default` but I think we read somewhere that we shouldn't use the default, so we changed them to `koku-metrics-manager-role`.

I'm fine with changing it to `controller-manager`, but would rather not if we didn't need to.

**Review by maskarb — COMMENTED** [review]
A couple of comments, but otherwise I think it looks good! Still need to test in a cluster though.

File: PROJECT [hunk 2]
```diff
@@ -2,9 +2,14 @@ domain: openshift.io
 layout: go.kubebuilder.io/v2
```
[the remark this answers is not in the document]
Comment (chambridge) [reply PROJECT:2]: updated

**Review by maskarb — COMMENTED** [review]
After testing bundle creation, I've found a series of additional changes we need to make:

Delete this [line](https://github.com/project-koku/koku-metrics-operator/blob/main/config/default/kustomization.yaml#L19) and this [folder](https://github.com/project-koku/koku-metrics-operator/tree/main/config/serviceaccounts).

Update the `namespace` [here](https://github.com/project-koku/koku-metrics-operator/blob/main/config/rbac/leader_election_role_binding.yaml#L12), [here](https://github.com/project-koku/koku-metrics-operator/blob/main/config/rbac/role_binding.yaml#L12), and [here](https://github.com/project-koku/koku-metrics-operator/blob/main/config/rbac/role_binding.yaml#L26) to match the namespace in the new service_account.yml (`system`)

After these changes are implemented, we should be good to go.

**Comment by chambridge** [comment]
@maskarb changes pushed

**Review by maskarb — APPROVED** [review]
Thanks for working through all these changes! LGTM!

**Pull request merged**